src/patronus/evals/evaluators.py

```diff
@@ -0,0 +1,252 @@
+            experiment_id=pat_logger._instrumentation_scope.attributes.get(Attributes.experiment_id),
+            evaluator_id=evaluator_id,
+            criteria=None,  # TODO
+            pass_=ev.pass_,
+            score=ev.score,
+            text_output=ev.text_output,
+            metadata=ev.metadata,
+            explanation=ev.explanation,
+            explanation_duration=ev.explanation_duration,
+            evaluation_duration=ev.evaluation_duration or duration,
+            metric_name=metric_name,
+            metric_description=metric_description,
+            dataset_id=None,  # TODO
+            dataset_sample_id=None,  # TODO
+            created_at=datetime.datetime.now(datetime.timezone.utc),
+            tags=ev.tags,
+            trace_id=hex(span_context.trace_id)[2:].zfill(32),
+            span_id=hex(span_context.span_id)[2:].zfill(16),
+        )
+        get_exporter().submit(eval_payload)
+    except Exception:
+        log.exception("Failed to submit evaluation payload to the exported")
+
+    # # Debug evaluation log
+    # ev_data = sorted([f"{k}={v!r}" for k, v in ev.model_dump().items() if v is not None])
+    # get_logger().warning(f"Evaluation({', '.join(ev_data)})")
+
+
+def coerce_eval_output_type(ev_output: typing.Any, class_name: Optional[str], func_name: str) -> EvaluationResult:
+    if isinstance(ev_output, EvaluationResult):
+        return ev_output
+    if isinstance(ev_output, bool):
+        return EvaluationResult(pass_=ev_output, score=float(ev_output))
+    if isinstance(ev_output, (int, float)):
+        return EvaluationResult(score=float(ev_output))
+    if isinstance(ev_output, str):
+        return EvaluationResult(text_output=ev_output)
+
+    e_name = func_name
+    if class_name:
+        e_name = f"{class_name}.{func_name}"
+
+    raise TypeError(
+        f"Evaluator function '{e_name}' returned unexpected type {type(ev_output)!r}. "
+        f"Supported return types are EvaluationResult, int, float, bool, str. "
+    )
+
+
+def evaluator(
+    *,
+    # Name for the evaluator. Defaults to function name (or class name in case of class based evaluators).
+    evaluator_id: Optional[str] = None,
+    # Name for the evaluation metric. Defaults to evaluator_id value.
+    metric_name: Optional[str] = None,
+    **kwargs,
+):
+    def decorator(fn):
+        fn_sign = inspect.signature(fn)
+        class_name: Optional[str] = kwargs.get("class_name")
+        fn_name: str = fn.__name__
+        eval_id: str = evaluator_id or fn_name
+        met_name: str = metric_name or eval_id
+        met_description: str = inspect.getdoc(fn)
+
+        @traced(ignore_input=True, ignore_output=True)
+        @functools.wraps(fn)
+        async def wrapper_async(*fn_args, **fn_kwargs):
+            start = time.perf_counter()
+            try:
+                ret = await fn(*fn_args, **fn_kwargs)
+            except Exception as e:
+                get_logger().exception("Evaluation failed")
```

Comment:
It would be nice to be able to pass exception object and add to log body / attributes.

Also i am not sure but if it's withing tracing wrepper, exceptions will be exported anyway 🤔